libvirt.go

```diff
@@ -717,3 +717,27 @@ func NewWithDialer(dialer socket.Dialer) *Libvirt {
 func New(conn net.Conn) *Libvirt {
 	return NewWithDialer(dialers.NewAlreadyConnected(conn))
 }
+
+// NetworkUpdateFixed is a wrapper over NetworkUpdate which swaps `Command` and `Section` when needed.
+// This function must be used instead of NetworkUpdate to be sure that the
+// NetworkUpdate call works both with older and newer libvirtd connections.
+//
+// libvirt on-wire protocol had a bug for a long time where Command and Section
+// were reversed. It's been fixed in newer libvirt versions, and backported to
+// some older versions. This helper detects what argument order libvirtd expects
+// and makes the correct NetworkUpdate call.
+func (l *Libvirt) NetworkUpdateFixed(Net Network, Command NetworkUpdateCommand, Section NetworkUpdateSection, ParentIndex int32, XML string, Flags NetworkUpdateFlags) (err error) {
```

Comment:
Just a suggestion, I don't think the current name is bad, but what do you think of "NetworkUpdateAdaptive" or "NetworkUpdateCompat" or something like that?

Reply:
`NetworkUpdateCompat` is much better name, thanks!